python/src/scipp/plot/plot.py

```diff
@@ -20,49 +20,105 @@ def __init__(self, *arg, **kw):
+    the plotting library.
     """
+    coords = {}
+    for dim, size in zip(variable.dims, variable.shape):
+        coords[dim] = make_fake_coord(dim, size)
+    return sc.DataArray(data=variable, coords=coords)
 
+
+def _raise_plot_input_error(intype):
+    """
+    Raise an error on bad input type.
+    """
+    raise RuntimeError("plot: Unknown input type: {}. Allowed inputs are "
+                       "a Dataset, a DataArray, a Variable (and their "
+                       "respective proxies), and a dict of "
+                       "Variables or DataArrays.".format(intype))
+
+
+def _parse_input(scipp_obj):
+    """
+    Parse plot input. Possible inputs are:
+      - Variable
+      - Dataset
+      - DataArray
+      - dict of Variables
+      - dict of DataArrays
+    Decompose the input and return a dict of DataArrays.
+    """
     inventory = dict()
     if su.is_dataset(scipp_obj):
         for name in sorted(scipp_obj.keys()):
             inventory[name] = scipp_obj[name]
     elif su.is_variable(scipp_obj):
-        coords = {}
-        for dim, size in zip(scipp_obj.dims, scipp_obj.shape):
-            coords[dim] = make_fake_coord(dim, size)
-        inventory[str(type(scipp_obj))] = sc.DataArray(data=scipp_obj,
-                                                       coords=coords)
+        inventory[str(type(scipp_obj))] = _variable_to_data_array(scipp_obj)
     elif su.is_data_array(scipp_obj):
         inventory[scipp_obj.name] = scipp_obj
     elif isinstance(scipp_obj, dict):
-        inventory = scipp_obj
+        for key in scipp_obj.keys():
+            if su.is_variable(scipp_obj[key]):
+                inventory[key] = _variable_to_data_array(scipp_obj[key])
+            elif su.is_data_array(scipp_obj[key]):
+                inventory[key] = scipp_obj[key]
+            else:
+                _raise_plot_input_error(type(scipp_obj[key]))
     else:
-        raise RuntimeError("plot: Unknown input type: {}. Allowed inputs are "
-                           "a Dataset, a DataArray, a Variable (and their "
-                           "respective proxies), and a dict of "
-                           "DataArrays.".format(type(scipp_obj)))
+        _raise_plot_input_error(type(scipp_obj))
+    return inventory
+
+
+def plot(scipp_obj,
+         projection=None,
+         axes=None,
+         color=None,
+         marker=None,
+         linestyle=None,
+         linewidth=None,
+         bins=None,
+         **kwargs):
+    """
+    Wrapper function to plot a scipp object.
+
+    Possible inputs are:
+      - Variable
+      - Dataset
+      - DataArray
+      - dict of Variables
+      - dict of DataArrays
+
+    1D Variables are grouped onto the same axes if they have the same dimension
+    and the same unit.
+
+    Any other Variables are displayed in their own figure.
+
+    Returns a Plot object which can be displayed in a Jupyter notebook.
+    """
```

Comment:
They are summarized well elsewhere but is there any worth in adding the information about the various input params here if this wrapper is what a user is most likely to see?

Reply:
The user would not actually see this, it would see the one in the `__init__.py` file, which has all the parameters listed.
When you do `help(plot)` you see the one with the long docstring